src/FakeItEasy/Creation/CreationResult.cs

```diff
@@ -60,13 +47,33 @@ public static CreationResult FailedToCreate(Type type, List<ResolvedConstructor>
             return new CreationResult(type, null, consideredConstructors);
         }
 
-        private string GetFailedToCreateResultMessage()
+        public object GetResultAsFake()

```

Comment:
I'm not sure about this approach. The caller could easily call `GetResultAsFake` instead of `GetResultAsDummy`, or vice versa (of course, the caller is us, since it's an internal class, but still...)
Could the `CreationResult` record whether it's for a fake or dummy creation instead? I think it's not very straightforward, since some dummies are actually fakes, but it would probably be better. Forget about it if it's too complex

Reply:
I had some of the same concerns, and even debated creating the `CreationResult` to know whether it was for a Fake or a Dummy, just as you suggest. I didn't mostly for the reason you mention: sometimes we start creating a Fake but actually intend to use it as a Dummy. Hmmm. I will quick try something to see how you like it. Stay tuned.

Reply:
Okay. I put one rough attempt at https://github.com/blairconrad/FakeItEasy/commit/30b9ec09a994277806e879d7fcbc0284d824f8f5.

Reply:
Not bad! But don't spend too much time on this, it's probably not worth it...

Reply:
I massaged that rough attempt a little bit and added it here as a fixup. Tell me what you think.